docs/source/tutorial/basic.rst

```diff
@@ -2,21 +2,629 @@
+
+A typical real-world example of a computer is a remote supercomputing facility.
+Codes can be anything from a Python script to powerful *ab initio* codes such as Quantum Espresso or machine learning tools like Tensorflow.
+Let's have a look at the codes that are available to us:
+
+.. code:: console
+
+    $ verdi code list
+    # List of configured codes:
+    # (use 'verdi code show CODEID' to see the details)
+    * pk 5 - add@tutor
+
+You can see a single code ``add@tutor``, with PK = 5, in the printed list.
+This code allows us to add two integers together.
+The ``add@tutor`` identifier indicates that the code with label ``add`` is run on the computer with label ``tutor``.
+To see more details about the computer, you can use the following ``verdi`` command:
+
+.. code:: console
+
+    $ verdi computer show tutor
+    Computer name:     tutor
+     * PK:             1
+     * UUID:           b9ecb07c-d084-41d7-b862-a2b1f02722c5
+     * Description:
+     * Hostname:       localhost
+     * Transport type: local
+     * Scheduler type: direct
+     * Work directory: /Users/mbercx/epfl/tutorials/my_tutor/work
+     * Shebang:        #!/bin/bash
+     * mpirun command: mpirun -np {tot_num_mpiprocs}
+     * prepend text:
+     # No prepend text.
+     * append text:
+     # No append text.
+
+We can see that the *Work directory* has been set up as the ``work`` subdirectory of the current directory.
+This is the directory in which the calculations running on the ``tutor`` computer will be executed.
+
+.. note::
+
+    You may have noticed that the PK of the ``tutor`` computer is 1, same as the ``Int`` node we created at the start of this tutorial.
+    This is because different entities, such as nodes, computers and groups, are stored in different tables of the database.
+    So, the PKs for each entity type are unique for each database, but entities of different types can have the same PK within one database.
+
+Let's now start up the ``verdi shell`` again and load the ``add@tutor`` code using its label:
+
+.. code-block:: ipython
+
+    In [1]: code = load_code(label='add')
+
+Every code has a convenient tool for setting up the required input, called the *builder*.
+It can be obtained by using the ``get_builder`` method:
+
+.. code-block:: ipython
 
-See `issue #3981 <https://github.com/aiidateam/aiida-core/issues/3981>`_.
+    In [2]: builder = code.get_builder()
 
+Using the builder, you can easily set up the calculation by directly providing the input arguments.
+Let's use the ``Int`` node that was created by our previous ``calcfunction`` as one of the inputs and a new node as the second input:
+
+.. code-block:: ipython
+
+    In [3]: builder.x = load_node(pk=4)
+       ...: builder.y = Int(5)
+
+In case your nodes' PKs are different, and you don't remember the PK of the output node from the previous calculation, you can also check the provenance graph you have generated and use the start of the UUID of the output node:
```

Comment:
```suggestion
In case that your nodes' PKs are different and you don't remember the PK of the output node from the previous calculation, check the provenance graph you generated earlier and use the UUID of the output node instead:
```